Automattic/newspack-plugin
# Add/apple pay file

### All Submissions:






### Changes proposed in this Pull Request:



A couple of changes to Stripe integration:

1. Exposing `is_currency_zero_decimal` method – this will be used by the Donate block in `newspack-blocks`
2. Adding necessary configuration for Apple Pay support – again this will be used by the Donate block
3. Fixing an issue where a new payment method was replacing the old one, instead of being added

### How to test the changes in this Pull Request:

1. **Payment method updating**
    1. On `master`, make two donations using different [test cards](https://stripe.com/docs/testing#cards) but the same email for the donor
    2. Observe the the donor has just one payment method on Stripe – the latest used card
    3. Switch to this branch and re-do step 1. Observe the donor now has two cards listed as payment methods 
1. **Apple Pay domain.** Configuring Apple Pay on Stripe requires two things: i) hosting a special file on the site, and ii) adding the domain on Stripe:
    1. Visit `<your-site>/.well-known/apple-developer-merchantid-domain-association` and observe that a file is downloaded
    1. A _definitive_ test would only be possible with a live Stripe account – let's skip that for the actual Apple Pay integration (coming up in `newspack-blocks`)

### Other information:
